src/erk/artifacts/preamble.py

```diff
@@ -0,0 +1,63 @@
+"""CLI preamble check for artifact staleness."""
+
+import sys
+from pathlib import Path
+
+import click
+
+from erk.artifacts.staleness import check_staleness, is_dev_mode
+
+
+def check_and_prompt_artifact_sync(project_dir: Path, no_sync: bool = False) -> None:
+    """Check for stale artifacts and prompt to sync.
+
+    Called from CLI callback before command execution.
+
+    Args:
+        project_dir: Project root directory
+        no_sync: If True, skip the staleness check entirely
+    """
+    if no_sync:
+        return
+
+    # Skip in dev mode - artifacts read from source
+    if is_dev_mode(project_dir):
+        return
+
+    result = check_staleness(project_dir)
+
+    if not result.is_stale:
+        return
+
+    # Skip silently if not initialized - artifacts are optional
+    # Users can run 'erk init' or 'erk artifacts sync' to enable artifact sync
+    if result.reason == "not initialized":
```

Comment:
hard cording to a string like this pass. reason should be a literal[str] (enum-esque) if we are going to do this